src/extension/gogoduck/src/main/java/au/org/emii/gogoduck/worker/GoGoDuckModule.java

```diff
@@ -0,0 +1,176 @@
+package au.org.emii.gogoduck.worker;
+
+import java.io.*;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import ucar.nc2.Attribute;
+import ucar.nc2.NetcdfFile;
+import ucar.nc2.NetcdfFileWriter;
+
+import au.org.emii.gogoduck.worker.GoGoDuck.UserLog;
+
+public class GoGoDuckModule {
+    private static final Logger logger = LoggerFactory.getLogger(GoGoDuckModule.class);
+
+    protected String profile = null;
+    protected String geoserver = null;
+    protected SubsetParameters subset = null;
+    protected UserLog userLog = null;
+
+    public GoGoDuckModule() {}
```

Comment:
There is a method which gets a Constructor for a Class based on the arguments for the constructor. Might help you do this if you wanted to give it a go.
https://docs.oracle.com/javase/8/docs/api/java/lang/Class.html#getConstructor-java.lang.Class...-